TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -61,6 +63,36 @@ public override bool BeginCodeElement(CodeElement codeElement)
             return false;
         }
 
+        public override bool Visit(End end)
+        {
+            // detect CodeElement with a mix of Debug and "Normal" lines in debugging mode         
+            if (DebuggingMode && end.CodeElement.Type != CodeElementType.SentenceEnd)
+            {
+                CodeElement openingCodeElement = end.Parent.CodeElement;
+                if (openingCodeElement != null)
+                {
+                    CodeElement endCodeElement = end.CodeElement;
+                    bool isStartDebug = IsCodeElementInDebug(openingCodeElement);
+                    bool isEndDebug = IsCodeElementInDebug(endCodeElement);
```

Comment:
@LANOYDO One things to know is that in cobol many constructions have optional END instruction, so optional END have no associated End node, thus this algorithm will never recognize something like.
````cobol
      d    evaluate xxxx
      d         when 'toto' display "found"
                when other display "not found"
```